File: PROGRAMMERS/LEVEL2/문자열 압축.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string s) {
    int answer = s.length();
    for (int i = 1; i <= s.length() / 2; i++) {
        string tmp = "";

        for (int j = 0; j < s.length(); j += i) {
            if (j + 2 * i <= s.length()) {
                if (s.substr(j, i) != s.substr(j + i, i)) {
                    tmp += s.substr(j, i);
                    continue;
                }
                tmp += s.substr(j, i);
                int cnt = 1;
                while (j + 2 * i <= s.length() && s.substr(j, i) == s.substr(j + i, i)) {
                    cnt++;
                    j += i;
                }
                tmp += to_string(cnt);
            }
            else {
                tmp += s.substr(j);
                break;
            }
        }
        if (answer > tmp.size())answer = tmp.size();
    }
    return answer;
}

File: PROGRAMMERS/LEVEL2/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = s.length();
        
        for(int i=1;i<=s.length()/2;i++) {
        	StringBuilder sb = new StringBuilder();
        	String cur = s.substring(0,i);
        	String next = "";
        	int cnt=1;
        	
        	for(int j=i;j<=s.length();j+=i) {
        		if(s.length()<=j+i) next = s.substring(j);
        		else next=s.substring(j,j+i);
        		
        		
        		if(cur.equals(next)) {
        			cnt++;
        		}
        		else if(cnt==1) {
        			sb.append(cur);
        			cur = next;
       			}
       			else {
       				sb.append(cnt).append(cur);
       				cur = next;
       				cnt=1;
       			}
       		}
        	 if(i!=cur.length()) sb.append(cur);
             	
             answer = Math.min(answer, sb.length());
       	}
       
        return answer;
    }
}